components/server/src/ome/logic/AdminImpl.java

```diff
@@ -1442,4 +1557,29 @@ private void adminOrPiOfNonUserGroups(ExperimenterGroup defaultGroup,
                 nonUserGroupGroups.toArray(new ExperimenterGroup[0]));
     }
 
+    /**
+     * @param privileges the privileges that are expected to be among those of the {@link AdminPrivilege} enumeration values
+     * @throws ApiUsageException if passed an unknown privilege
+     */
+    private void assertKnownPrivileges(Iterable<AdminPrivilege> privileges) {
+        for (final AdminPrivilege privilege : privileges) {
+            if (adminPrivileges.getPrivilege(privilege.getValue()) == null) {
+                throw new ApiUsageException("unknown light administrator privilege: " + privilege.getValue());
+            }
+        }
+    }
+
+    /**
+     * @return the light administrator privileges that apply to the current session
+     */
+    private Set<AdminPrivilege> getCurrentAdminPrivilegesForSession() {
+        if (isAdmin()) {
+            final String hql = "FROM Session s LEFT OUTER JOIN FETCH s.sudoer WHERE s.id = :id";
+            final Parameters params = new Parameters().addId(getEventContext().getCurrentSessionId());
+            final ome.model.meta.Session session = iQuery.findByQuery(hql, params);
```

Comment:
Has there been an performance impact by querying for the privileges? If so, we should likely re-use the `SessionCache` infrastructure to pre-load them.